lms/services/mailchimp.py

```diff
@@ -40,6 +42,66 @@ def __init__(self, db, api_key):
         self.db = db
         self.mailchimp_client = mailchimp_transactional.Client(api_key)
 
+    def send(  # pylint:disable=too-many-arguments
+        self,
+        template: str,
+        sender: EmailSender,
+        recipient: EmailRecipient,
+        template_vars: dict,
+        unsubscribe_url: Optional[str] = None,
+        task_done_key: Optional[str] = None,
+    ):
+        """
+        Send an email using Mailchimp Transactional's API.
+
+        https://mailchimp.com/developer/transactional/api/messages/send-new-message/
+        """
+
+        if task_done_key:
+            if self.db.execute(
+                select(TaskDone).filter_by(key=task_done_key)
+            ).one_or_none():
+                LOG.info("Not sending duplicate email %s", task_done_key)
+                return
+
+        headers = {}
+
+        if unsubscribe_url:
+            template_vars["unsubscribe_url"] = unsubscribe_url
+            headers["List-Unsubscribe"] = unsubscribe_url
+
+        subject = render(str(template / Path("subject.jinja2")), template_vars)
+
+        params = {
+            "message": {
+                "subject": subject,
+                "html": "...",  # For logging only, will be replaced below.
+                "subaccount": sender.subaccount,
+                "from_email": sender.email,
+                "from_name": sender.name,
+                "to": [{"email": recipient.email, "name": recipient.name}],
+                "track_opens": True,
+                "track_clicks": True,
+                "auto_text": True,
+                "headers": headers,
+            },
+            "async": True,
+        }
+
+        LOG.info("mailchimp_client.send(%r)", params)
+
+        # The HTML body might be long so add it to the params *after* logging.
+        params["message"]["html"] = render(str(template / Path("body.html.jinja2")), template_vars)
```

Comment:
This is mostly the same Mailchimp API params as `send_template()`a below. Notes:

* We're calling `render()` to render our new Jinja2 subject and body templates, and passing the resulting text to Mailchimp
* We're using Mailchimp's `"auto_text": True` to have it automatically generate the plain text version of the email from the HTML for us. We could instead render a separate `body.txt.jinja2` template ourselves which could make for a more finely-crafted plain text version but that template is also a bother to create and maintain. Perhaps for a follow-up PR
* I'm doing some faff to avoid logging the entire HTML body to the terminal because it's massive